crates/router/src/core/payments/types.rs

```
@@ -417,7 +417,10 @@ impl ForeignTryFrom<&router_request_types::authentication::AuthenticationStore>
                 .change_context(errors::ApiErrorResponse::InternalServerError)
                 .attach_printable("cavv must not be null when authentication_status is success")?;
             Ok(Self {
-                eci: authentication.eci.clone(),
+                eci: authentication
+                    .eci
+                    .as_ref()
+                    .and_then(|s| common_enums::enums::Eci::from_str(s).ok()),
```

why are we ignoring the error here, also if we donot want to propogate the error can we `inspect` and log the error